### PR TITLE
Feature Flags: use FeatureToggles interface where possible

### PR DESCRIPTION
**Pull request by hairyhenderson**
**What is this feature?**

Refactoring to reference `*featuremgmt.FeatureManager` outside the `featuremgmt` package as sparingly as possible, with the goal of making it simpler to potentially replace it with a different implementation later on.

A large number of references were from unit tests - these were all replaced by a test implementation.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/expr/service_test.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ func TestDSQueryError(t *testing.T) {
 		cfg:          setting.NewCfg(),
 		dataService:  me,
 		pCtxProvider: pCtxProvider,
-		features:     &featuremgmt.FeatureManager{},
+		features:     featuremgmt.TestFeatureToggles(),
```
Comment (ryantxu) [inline pkg/expr/service_test.go:141]:
```suggestion
		features:     featuremgmt.WithFeatures(),
```
WithFeatures is used in many of the other tests -- it lets you stick in features if you want them enabled.  I don't think we need another test implementation (maybe an alias?)

Comment (hairyhenderson) [reply pkg/expr/service_test.go:141]: Ah, thanks @ryantxu - I missed this. I'll replace it!

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Review by IevaVasiljeva — APPROVED** [review]
LGTM for auth related changes. Thanks for cleaning this up!

**Comment by hairyhenderson** [comment]
hm... @papagian after my first rebase the CI worked, but I rebased again (mostly to make the `license/cla` test run again), and now that same test is failing again 😞

**Comment by papagian** [comment]
> hm... @papagian after my first rebase the CI worked, but I rebased again (mostly to make the `license/cla` test run again), and now that same test is failing again 😞

ok, that was slightly different: I have another [fix](https://github.com/grafana/grafana/pull/85408)🤞

**Review by jalevin — APPROVED** [review]

**Pull request merged**